### PR TITLE
chore: reduce warnings

### PR DESCRIPTION
**Pull request by languitar**
This PR tries to get rid of some of the easy to solve compiler warnings. With all the warnings, it becomes more and more tedious to actually spot real issues in the logs.

### CODE REVIEW

File: model-server/src/test/kotlin/org/modelix/model/server/ReplicatedRepositoryTest.kt [hunk 1]
```diff
@@ -437,9 +439,9 @@ class ReplicatedRepositoryTest {
             nonMergedVersions.remove(versionToMerge)
         }
 
-        for (i in 1..100) {
+        (1..100).forEach {
```
Comment (mhuster23) [inline model-server/src/test/kotlin/org/modelix/model/server/ReplicatedRepositoryTest.kt:442]: We should probably keep the regular for loop to conform with the style guide.

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/HistoryHandler.kt [hunk 2]
```diff
@@ -169,7 +178,14 @@ class HistoryHandler(val client: IModelClient, private val repositoriesManager:
         skip: Int,
         limit: Int,
     ) {
-        val headVersion = if (headHash == null || headHash.length == 0) latestVersion else CLVersion(headHash, client.storeCache!!)
+        val headVersion = if (headHash == null || headHash.length == 0) {
```
Comment (mhuster23) [inline model-server/src/main/kotlin/org/modelix/model/server/handlers/HistoryHandler.kt:181]: `headHash.isNullOrEmpty()`

File: model-server/src/test/kotlin/org/modelix/model/server/ReplicatedRepositoryTest.kt (hunk 1 above)
Comment (languitar) [reply model-server/src/test/kotlin/org/modelix/model/server/ReplicatedRepositoryTest.kt:442]: Right, the change was older than our discussion about the style guide. I am still confused about that rule ;-)

Comment (languitar) [reply model-server/src/test/kotlin/org/modelix/model/server/ReplicatedRepositoryTest.kt]: Replaced with `repeat`, which is hopefully even more readable now.

File: model-server/src/test/kotlin/org/modelix/model/server/ReplicatedRepositoryTest.kt [hunk 3]
```diff
@@ -108,7 +108,9 @@ class ReplicatedRepositoryTest {
             }
             branchToChange.runWrite {
                 val changeGenerator = RandomModelChangeGenerator(branchToChange.getRootNode(), rand)
-                for (i in 1..1000) changeGenerator.applyRandomChange()
+                (1..1000).forEach { _ ->
```
Comment (mhuster23) [inline model-server/src/test/kotlin/org/modelix/model/server/ReplicatedRepositoryTest.kt:111]:
```suggestion
                repeat(1000) { _ ->
```

**Review by mhuster23 — APPROVED** [review]

**Pull request merged**